openg2p_registration/models/openg2p_registration.py

```diff
@@ -346,17 +369,89 @@ def create_beneficiary_from_registration(self):
             'context': context
         }
 
+    @api.onchange('stage_id')
+    def checking_duplicates(self):
+        if int(self.stage_id.id) == 2:
+            r = self.search_beneficiary()
+            if r == True:
+                self.update({'duplicate_beneficiaries_ids': (self.beneficiary_id)
```

Comment:
Over here you're doing self.beneficiary_id but the beneficiary is is coming from response of the previous call?